### ISSUE TITLE
Monitor TCP traffic

### ISSUE THREAD

**Issue by CorentinWicht** [LostRuins/koboldcpp#665]
Dear Developper,

At our University, we are running Koboldcpp on a server accessible within our network and we would like to monitor incoming TCP traffic  to localhost port 8008 (to estimate the interest in our AI local implementation service).

Would there a possibility to extract some logs within Koboldcpp to have a broad estimate or daily usage (maybe by simply counting the number of unique IP addresses) ?

For your information our koboldcpp instance is constantly running as a background job from within a service on a Redhat 9 server:
```
[Unit]
Description=Run script at startup after network becomes reachable
After=network.target
[Service]
Type=simple
RemainAfterExit=yes
ExecStart=/usr/bin/python /opt/koboldcpp-latest/koboldcpp.py /home/frillm/Models/llama-2-70b-chat.Q5_K_M.gguf 8008 --multiuser 20 --highpriority --usecublas --gpulayers 79 --nommap --preloadstory /opt/settings.json
EnvironmentFile=/etc/koboldai-env.conf
TimeoutStartSec=0

[Install]
WantedBy=default.target
```

Best wishes,

C.

**Comment by LostRuins** [comment]
No need to monitor traffic. You can do even better. 

KoboldCpp comes with an inbuilt performance endpoint that provides performance stats, one of the measured stats is number of generations completed. You can retrieve this information at `/api/extra/perf/`

If there are other metrics that you'd like to capture, let me know.

For API reference: 
https://lite.koboldai.net/koboldcpp_api#/extra/get_extra_perf

**Comment by CorentinWicht** [comment]
> No need to monitor traffic. You can do even better.
> 
> KoboldCpp comes with an inbuilt performance endpoint that provides performance stats, one of the measured stats is number of generations completed. You can retrieve this information at `/api/extra/perf/`
> 
> If there are other metrics that you'd like to capture, let me know.
> 
> For API reference: https://lite.koboldai.net/koboldcpp_api#/extra/get_extra_perf

Dear Developper,

Many thanks for your suggestion, I can confirm that it's very close to what we were looking for.

Though, since we are running KoboldCpp on a RHEL9 server accessible within our University network to all personnel/students (>10000 potential users), the performance stats (and especially the number of generations) that I can retrieve at `/api/extra/perf/` seems to only correspond to the requests that I personnally send (in my own "Session"). Would there be a way to monitor the number of generations across all user-specific Sessions, for a given day?  

Basically, we are just trying to measure the interest in our local LLM implementation compared to other widely used alternatives.

Thanks,

C.

**Comment by LostRuins** [comment]
Hmm I don't understand, KoboldCpp is a standalone server. The stats you see apply to anyone who uses that server. Are you saying you run multiple different instances of KoboldCpp rather than a shared one? 

For a single koboldcpp server the stats apply to anyone who connects to it, regardless of device or browser.

**Comment by CorentinWicht** [comment]
> Hmm I don't understand, KoboldCpp is a standalone server. The stats you see apply to anyone who uses that server. Are you saying you run multiple different instances of KoboldCpp rather than a shared one?
> 
> For a single koboldcpp server the stats apply to anyone who connects to it, regardless of device or browser.

My bad, you are absolutely correct. I just realized that at each server reboot the API stats are also reset and that led me to wrong conclusions.
